controllers/workspace_controller_variable_sets_test.go

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllers
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	tfc "github.com/hashicorp/go-tfe"
+	appv1alpha2 "github.com/hashicorp/hcp-terraform-operator/api/v1alpha2"
+)
+
+var _ = Describe("Workspace controller", Ordered, func() {
```

Comment:
All in all, it's great that you've added tests here. However, they are failing due to the lack of connection between workspace reconciliation and the new logic you're working on. Please refer to my comment in the `controllers/workspace_controller_variable_sets.go` file.

You can run tests locally and focus only on a specific one by passing an addition argument:

```console
$ make test TESTARGS='-ginkgo.focus "VariableSet"'
```

You need to export the following environment variables to make them run:

```console
$ export TFC_TOKEN=<TFC_API_TOKEN>
$ export TFC_ORG=<TFC_ORG_NAME>
```

Reply:
This is still an issue with the latest commit.